internal/handlers/common/aggregations/projection.go

```diff
@@ -69,6 +69,14 @@ func ValidateProjection(projection *types.Document) (*types.Document, bool, erro
 			)
 		}
 
+		if strings.HasSuffix(key, "$") {
+			return nil, false, commonerrors.NewCommandErrorMsgWithArgument(
+				commonerrors.ErrAggregatePositionalProject,
+				"Invalid $project :: caused by :: Cannot use positional projection in aggregation projection",
+				"$project (stage)",
+			)
+		}
```

Comment:
This if statement works for `foo.$` but it also works for `foo$` which is a valid key in mongodb

Reply:
Yeah great catch! It should work, added test for that

Reply:
See this test case https://github.com/FerretDB/FerretDB/blob/6535691d4fdad23d25466243bbfefcf9abf3e321/integration/query_projection_compat_test.go#L277